router/internal/retrytransport/retry_transport.go

```diff
@@ -101,6 +101,12 @@ func (rt *RetryHTTPTransport) RoundTrip(req *http.Request) (*http.Response, erro
 	return resp, err
 }
 
+func isResponseOK(resp *http.Response) bool {
+	// Ensure we don't wait for no reason when subgraphs don't behave
+	// spec-compliant and returns a different status code than 200.
+	return resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices
```

Comment:
Can you add a comment that this is all status 2XX?

Maybe a better name of the function is `isResponse2XX`?

Reply:
I'm going to use numbers.

Reply:
For me, response OK is 200 and 200 only. The name is misleading. I'd choose `isResponse2XX`, which is what the function actually does. But I won't push it further. :D